back/src/controllers/cra.controller.ts

```diff
@@ -0,0 +1,113 @@
+import { CraApplication } from "../domain/application/craApplication";
+import { CreateAbsenceDto } from "../Dto/CreateAbsenceDto";
+import { Absence } from "../domain/model/Absence";
+import { Body, Controller, Delete, Get, HttpException, HttpStatus, Param, Post } from "@nestjs/common";
+import { CreateActivityDto } from "../Dto/CreateActivityDto";
+import { Activity } from "../domain/model/Activity";
+import { deleteActivityAbsenceDto } from "@app/Dto/deleteActivityAbsenceDto";
+
+@Controller('cra')
+export class CraController {
+  constructor(private readonly craApp: CraApplication) { }
+  /*
+      @Post()
+      async createCra(@Body() createCraDto: CreateCraDto): Promise<CRA> {
+        try {
+          const createdCra = await this.craService.createCra(createCraDto);
+          return createdCra;
+        } catch (error) {
+          console.error('Error creating CRA:', error);
+          throw new Error('Failed to create CRA');
+        }
+      }
+      */
+
+  @Post('absence')
+  async addAbsence(@Body() createAbsenceDto: CreateAbsenceDto): Promise<Absence> {
+    try {
+      const result = await this.craApp.addAbsence(createAbsenceDto);
+      return result;
+    }  catch (error) {
+      if (error.message.includes('it is a holiday')) {
```

Comment:
Do not discriminate the errors by the error message but by type using "instance of"